### ISSUE TITLE
[FN 3.5.7] No port other than 7770 is working in headless built game on linux

### ISSUE THREAD

**Issue by tedbarth** [FirstGearGames/FishNet#314]
Unity version: 2021 LTS and 2022 something
Fish-Networking version: 3.5.7

**Describe the bug**
When building headless server, the client cannot connect to ports other than 7770, at least on linux.

**To Reproduce**
1. Add NetworkController MonoBehaviour attached to this issue
2. Set _ServerManager's_ `StartOnHeadless` to _false_
3. Build a headless dedicated server using Unity Editor
4. Build a normal client using Unity Editor
5. Start server from Command line via `game.x86_64 --server --port 7770 --serverBindAddressIpv4 127.0.0.1`
6. Start client from Command line via `game.x86_64 --client --port 7770 --clientAddress 127.0.0.1`
7. Client can connect
8. Kill both, server and client
9. Start server from Command line via `game.x86_64 --server --port 7771 --serverBindAddressIpv4 127.0.0.1`
10. Start client from Command line via `game.x86_64 --client --port 7771 --clientAddress 127.0.0.1`
11. Client can't connect

**Expected behavior**
The client can connect to the server no matter what port is used and if headless mode.

**Code**
```
using System;
using FishNet;
using FishNet.Transporting;
using FishNet.Transporting.Tugboat;
using UnityEngine;

namespace Networking {
  public class NetworkController : MonoBehaviour {
    private void Awake() {
      var tugboat = InstanceFinder.NetworkManager.gameObject.GetComponent<Tugboat>();

      bool startServer = false;
      bool startClient = false;
      ushort port = tugboat.GetPort();
      string clientAddress = tugboat.GetClientAddress();
      string serverBindAddressIpv4 = tugboat.GetServerBindAddress(IPAddressType.IPv4);
      string serverBindAddressIpv6 = tugboat.GetServerBindAddress(IPAddressType.IPv6);

      Debug.Log("Parsing commandline arguments...");
      string[] args = Environment.GetCommandLineArgs();
      for (int i = 0; i < args.Length; i++) {
        var argument = args[i];
        switch (argument) {
          case "--server":
            startServer = true;
            break;
          case "--client":
            startClient = true;
            break;
          case "--clientAddress":
            clientAddress = args[++i];
            break;
          case "--serverBindAddressIpv4":
            serverBindAddressIpv4 = args[++i];
            break;
          case "--serverBindAddressIpv6":
            serverBindAddressIpv6 = args[++i];
            break;
          case "--port":
            port = ushort.Parse(args[++i]);
            break;
        }
      }

      Debug.Log(
        $"Parsing commandline arguments ended:\n"
        + $"Start server: {startServer}\n"
        + $"Start client: {startClient}\n"
        + $"Client address: {clientAddress}\n"
        + $"Server bind address IPv4: {serverBindAddressIpv4}\n"
        + $"Server bind address IPv6: {serverBindAddressIpv6}\n"
        + $"Port: {port}");

      tugboat.SetServerBindAddress(serverBindAddressIpv4, IPAddressType.IPv4);
      tugboat.SetServerBindAddress(serverBindAddressIpv6, IPAddressType.IPv6);
      tugboat.SetClientAddress(clientAddress);
      tugboat.SetPort(port);

      if (startServer) {
        StartServer();
      }

      if (startClient) {
        StartClient();
      }
    }

    private static void StartClient() {
      Debug.Log("Starting Client...");
      InstanceFinder.ClientManager.StartConnection();
    }

    private static void StartServer() {
      Debug.Log("Starting Server...");
      InstanceFinder.ServerManager.StartConnection();
    }
  }
}
```

**Comment by tedbarth** [comment]
While testing I also see this message when starting the server for brand new unused ports:
```
[B]Bind exception: System.Net.Sockets.SocketException (0x80004005): Address already in use
  at System.Net.Sockets.Socket.Bind (System.Net.EndPoint localEP) [0x00043] in <b03d1082af2d4d4290a36c0a5a3293bc>:0 
  at LiteNetLib.NetManager.BindSocket (System.Net.Sockets.Socket socket, System.Net.IPEndPoint ep) [0x00169] in /home/barth/nobackup/private/projects/monochrome-unity/game/Assets/FishNet/Runtime/Transporting/Transports/Tugboat/LiteNetLib/NetManager.Socket.cs:392 , errorCode: AddressAlreadyInUse
[S] Cannot restore connection. Ipv4 127.0.0.1, Ipv6 ::, Port 17770, ManualMode False
```

The port list before doesn't have port 17770 before starting server:
```
$ netstat -tulpn
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:6463          0.0.0.0:*               LISTEN      34607/Discord --typ 
tcp        0      0 127.0.0.1:36479         0.0.0.0:*               LISTEN      44532/Rider.Backend 
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:33851         0.0.0.0:*               LISTEN      44532/Rider.Backend 
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      -                   
tcp6       0      0 :::22                   :::*                    LISTEN      -                   
tcp6       0      0 :::39000                :::*                    LISTEN      32929/unityhub-bin  
tcp6       0      0 127.0.0.1:52829         :::*                    LISTEN      20360/./jetbrains-t 
tcp6       0      0 127.0.0.1:63342         :::*                    LISTEN      44377/java          
tcp6       0      0 ::1:631                 :::*                    LISTEN      -                   
tcp6       0      0 127.0.0.1:37087         :::*                    LISTEN      44377/java          
udp        0      0 127.0.0.53:53           0.0.0.0:*                           -                   
udp        0      0 0.0.0.0:631             0.0.0.0:*                           -                   
udp        0      0 0.0.0.0:37438           0.0.0.0:*                           -                   
udp        0      0 224.0.0.251:5353        0.0.0.0:*                           21158/chromium --en 
udp        0      0 224.0.0.251:5353        0.0.0.0:*                           21197/chromium --ty 
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           -                   
udp6       0      0 :::33091                :::*                                -                   
udp6       0      0 fe80::eeb3:b3cc:9ad:546 :::*                                -                   
udp6       0      0 :::5353                 :::*                                -                  
```

But it has while running the server:
```
$ netstat -tulpn
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:6463          0.0.0.0:*               LISTEN      34607/Discord --typ 
tcp        0      0 127.0.0.1:36479         0.0.0.0:*               LISTEN      44532/Rider.Backend 
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:33851         0.0.0.0:*               LISTEN      44532/Rider.Backend 
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      -                   
tcp6       0      0 :::22                   :::*                    LISTEN      -                   
tcp6       0      0 :::39000                :::*                    LISTEN      32929/unityhub-bin  
tcp6       0      0 127.0.0.1:52829         :::*                    LISTEN      20360/./jetbrains-t 
tcp6       0      0 127.0.0.1:63342         :::*                    LISTEN      44377/java          
tcp6       0      0 ::1:631                 :::*                    LISTEN      -                   
tcp6       0      0 127.0.0.1:37087         :::*                    LISTEN      44377/java          
udp        0      0 127.0.0.53:53           0.0.0.0:*                           -                   
udp        0      0 0.0.0.0:631             0.0.0.0:*                           -                   
udp        0      0 0.0.0.0:37438           0.0.0.0:*                           -                   
udp        0      0 224.0.0.251:5353        0.0.0.0:*                           21158/chromium --en 
udp        0      0 224.0.0.251:5353        0.0.0.0:*                           21197/chromium --ty 
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           -                   
udp        0      0 127.0.0.1:17770         0.0.0.0:*                           65410/test-server.x 
udp6       0      0 :::33091                :::*                                -                   
udp6       0      0 fe80::eeb3:b3cc:9ad:546 :::*                                -                   
udp6       0      0 :::5353                 :::*                                -
```

I could not see a link to the bug, though. This also happens only for headless mode on *two different systems*:
* My local Ubuntu machine
* A linux Virtual Server I rent exactly for this purpose

**Comment by FirstGearGames** [comment]
Resolved in 3.5.8

**Comment by tedbarth** [comment]
The problem indeed is fixed. I uploaded an old version to the server. My bad, sorry for the hassle.
It works now, also on a remote server with different port! Thanks a lot!
